packages/xo-web/src/common/xo/add-hosts-modal/index.js

```diff
@@ -41,62 +41,62 @@ import { SelectHost } from 'select-objects'
   }),
   { withRef: true }
 )
-export default class AddHostModal extends BaseComponent {
+export default class AddHostsModal extends BaseComponent {
   get value() {
-    const { nHostMissingPatches, nPoolMissingPatches } = this.state
+    const { nHostsMissingPatches, nPoolMissingPatches } = this.state
     if (
       process.env.XOA_PLAN < 2 &&
-      (nHostMissingPatches > 0 || nPoolMissingPatches > 0)
+      (nHostsMissingPatches > 0 || nPoolMissingPatches > 0)
     ) {
       return {}
     }
 
-    return { host: this.state.host }
+    return { hosts: this.state.hosts }
   }
 
   _getHostPredicate = createSelector(
     () => this.props.singleHosts,
     singleHosts => host => singleHosts[host.id]
   )
 
-  _onChangeHost = async host => {
-    if (host === null) {
+  _onChangeHosts = async hosts => {
+    if (isEmpty(hosts)) {
       this.setState({
-        host,
-        nHostMissingPatches: undefined,
+        hosts,
+        nHostsMissingPatches: undefined,
         nPoolMissingPatches: undefined,
       })
       return
     }
 
-    const { master } = this.props.pool
-    const hostMissingPatches = await getPatchesDifference(host.id, master)
-    const poolMissingPatches = await getPatchesDifference(master, host.id)
+    const { pool } = this.props
+    const hostsMissingPatches = await getHostsMissingPatches(pool, hosts)
+    const poolMissingPatches = await getPoolMissingPatches(pool, hosts)
 
     this.setState({
-      host,
-      nHostMissingPatches: hostMissingPatches.length,
-      nPoolMissingPatches: poolMissingPatches.length,
+      hosts,
+      nHostsMissingPatches: flatMap(hostsMissingPatches).length,
+      nPoolMissingPatches: flatMap(poolMissingPatches).length,
     })
   }
 
   render() {
-    const { nHostMissingPatches, nPoolMissingPatches } = this.state
-
+    const { hosts, nHostsMissingPatches, nPoolMissingPatches } = this.state
     return (
       <div>
         <SingleLineRow>
-          <Col size={6}>{_('addHostSelectHost')}</Col>
+          <Col size={6}>{_('addHostsSelectHosts')}</Col>
           <Col size={6}>
             <SelectHost
-              onChange={this._onChangeHost}
+              multi
+              onChange={this._onChangeHosts}
               predicate={this._getHostPredicate()}
-              value={this.state.host}
+              value={hosts}
             />
           </Col>
         </SingleLineRow>
         <br />
-        {(nHostMissingPatches > 0 || nPoolMissingPatches > 0) && (
+        {(nHostsMissingPatches > 0 || nPoolMissingPatches > 0) && (
```

Comment:
I don't know if I understood what you mean but here the `nHostsMissingPatches`  is the number of different patches between the different hosts and the pool so it can be 0 but this isn't a case to display warning message.